### PR TITLE
Feat/tz aware dta

### LINKED ISSUE #2000: Allow different timezones in datetime_attribute_timeseries, other funcs and encoders

### PR DESCRIPTION
**Pull request by dennisbader**
fixes #2000.

### Summary
- adds support for time index time zone conversion with parameter `tz` before generating/computing holidays and datetime attributes.
  - `darts.utils.timeseries_generation.datetime_attribute_timeseries`
  - `darts.utils.timeseries_generation.holidays_timeseries`
  - all time axis encoders and `add_encoders` dict for automatic encoders in forecasting models
  - `darts.Timeseries.add_holidays`
  -` darts.Timeseries.add_datetime_attribute`

### CODE REVIEW

File: darts/tests/utils/test_timeseries_generation.py [hunk 1]
```diff
@@ -321,3 +344,86 @@ def test_calculation(coef):
...
+                pd.RangeIndex(start=0, stop=len(idx)), "h", vals_exp=np.arange(len(idx))
+            )
+
+        # invalid attribute
+        with pytest.raises(ValueError):
+            helper_routine(idx, "h", vals_exp=np.arange(len(idx)))
+
+        # no time zone aware index
+        with pytest.raises(ValueError):
+            helper_routine(idx.tz_localize("UTC"), "h", vals_exp=np.arange(len(idx)))
```
Comment (madtoinou) [inline darts/tests/utils/test_timeseries_generation.py:372]: this test could fail just because of the invalid attribute

File: darts/utils/timeseries_generation.py [hunk 2]
```diff
@@ -546,39 +547,46 @@ def holidays_timeseries(
     Parameters
     ----------
     time_index
-        The time index over which to generate the holidays
+        Either a `pd.DatetimeIndex` or a `TimeSeries` for which to generate the holidays/
```
Comment (madtoinou) [inline darts/utils/timeseries_generation.py:550]: Typo

```suggestion
        Either a `pd.DatetimeIndex` or a `TimeSeries` for which to generate the holidays
```

File: darts/utils/timeseries_generation.py [hunk 3]
```diff
@@ -546,39 +547,46 @@ def holidays_timeseries(
...
+        The state.
     until
         Extend the time_index up until timestamp for datetime indexed series
         and int for range indexed series, should match or exceed forecasting window.
     add_length
         Extend the time_index by add_length, should match or exceed forecasting window.
         Set only one of until and add_length.
     column_name
-        Optionally, the name of the value column for the returned TimeSeries
+        Optionally, the name of the value column for the returned TimeSeries/
```
Comment (madtoinou) [inline darts/utils/timeseries_generation.py:564]: Typo

```suggestion
        Optionally, the name of the value column for the returned TimeSeries
```

File: darts/dataprocessing/encoders/encoders.py [hunk 4]
```diff
@@ -837,6 +883,9 @@ def __init__(
             <darts.dataprocessing.transformers.fittable_data_transformer.FittableDataTransformer>` such as Scaler() or
             BoxCox(). The transformers will be fitted on the training dataset when calling calling `model.fit()`.
             The training, validation and inference datasets are then transformed equally.
+        Supported time zone:
+            optionally, apply a time zone conversion with keyword 'tz'. This converts the time zone-naive index to a
```
Comment (madtoinou) [inline darts/dataprocessing/encoders/encoders.py:887]:
```suggestion
            Optionally, apply a time zone conversion with keyword 'tz'. This converts the time zone-naive index to a
```

**Review by madtoinou — APPROVED** [review]
Nice! Just a few typo here and there.
